src/test/java/edu/cornell/kfs/fp/document/service/impl/CuDisbursementVoucherTaxServiceImplTest.java

```diff
@@ -0,0 +1,291 @@
+package edu.cornell.kfs.fp.document.service.impl;
+
+import java.util.Collection;
+import java.util.List;
+
+import junit.framework.TestCase;
+
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorContract;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+import org.kuali.kfs.vnd.businessobject.VendorRoutingComparable;
+import org.kuali.kfs.vnd.document.service.VendorService;
+import org.kuali.rice.core.api.parameter.ParameterEvaluator;
+import org.kuali.rice.core.api.parameter.ParameterEvaluatorService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.krad.bo.Note;
+import org.kuali.rice.krad.document.Document;
+
+public class CuDisbursementVoucherTaxServiceImplTest extends TestCase {
+	
+	private static final Integer FOREIGN_VENDOR_HEADER_ID = 1000;	
+	private static final Integer NON_FOREIGN_VENDOR_HEADER_ID = 2000;
+	private static final String PAYMENT_REASON_THAT_REQUIRES_TAX_REVIEW_FOR_FOREIGN_VENDOR = "Z";
+	private static final String PAYMENT_REASON_THAT_DOES_NOT_REQUIRE_TAX_REVIEW_FOR_FOREIGN_VENDOR = "F";
+	
+	private CuDisbursementVoucherTaxServiceImpl disbursementVoucherTaxService;
+	
+	@Override
+	protected void setUp() throws Exception {
+		super.setUp();
+		disbursementVoucherTaxService = new CuDisbursementVoucherTaxServiceImpl();
+		disbursementVoucherTaxService.setVendorService(new MockVendorService());
+		disbursementVoucherTaxService.setParameterEvaluatorService(new MockParameterEvaluatorService());	
+	}
+	
+	public void testIsForeignVendorWithTaxReviewRequired_true()  {
```

Comment:
You may need to rename these various test methods if you end up changing the CuDisbursementVoucherTaxService method names.